app.js

```diff
@@ -25,6 +25,10 @@ async function buildJob({
     jenkinsClient.job.build.bind(jenkinsClient.job),
   )(buildOptions);
 
+  if (!waitForEnd && failOnFailure) {
```

Comment:
Is this condition correct? The plugin will print `Waiting for build end to determine SUCCESS/FAILURE status.` when `waitForEnd` is `false` and `failOnFailure` is true

Reply:
Yes, it's correct. failOnFailure can only work correctly if we waitForEnd. The message is for the bonehead who wants to fail on failure but not wait on end. - sorry, but we have to wait for end in order to determine status.